src/waveform/renderers/waveformrendermarkrange.cpp

```diff
@@ -64,6 +64,13 @@ void WaveformRenderMarkRange::draw(QPainter *painter, QPaintEvent * /*event*/) {
 
         double startPosition = m_waveformRenderer->transformSamplePositionInRendererWorld(startSample);
         double endPosition = m_waveformRenderer->transformSamplePositionInRendererWorld(endSample);
+        double span = endPosition - startPosition;
+        // If span is < 1 painter would extend the rectangle to width() as long as
+        // startPosition is in view.
+        // TODO(xxx) Figure what's wrong with QPainter / the renderer
+        if (span < 1.0) {
+            span = 1.0;
+        }
```

Comment:
```suggestion
        // If span is < 1 painter would extend the rectangle to width() as long as
        // startPosition is in view.
        // TODO(xxx) Figure what's wrong with QPainter / the renderer
        const double span = std::max(endPosition - startPosition, 1.0);
```
improves constness, but requires an `#include <algorithm>`